libs/geoservices/covid/src/lib/pages/maps/components/lockdown-map/lockdown-map.component.ts

```diff
@@ -0,0 +1,258 @@
+      );
+
+      map.on('mousemove', 'counties-lockdowns', (e) => {
+        const feature = e.features[0];
+        this.countyBoxModel.next({
+          county: feature.properties.NAME
+        });
+      });
+
+      map.on('click', 'counties-lockdowns', (e) => {
+        const feature = e.features[0];
+        const hoveredCounty = this.statData.filter((county) => county.fips === feature.properties.fips);
+        const updatedDate: Date = hoveredCounty[0]['lockdownInfo'][0]
+          ? hoveredCounty[0]['lockdownInfo'][0]['updated']
+          : null;
+        const createdDate: Date = hoveredCounty[0]['lockdownInfo'][0]
+          ? hoveredCounty[0]['lockdownInfo'][0]['created']
+          : null;
+        this.infoBoxModel.next({
+          name: feature.properties.NAME,
+          state: hoveredCounty[0]['state'],
+          fips: hoveredCounty[0]['fips'],
+          fipsNum: hoveredCounty[0]['fipsNum'],
+          claims: hoveredCounty[0]['claims'],
+          sites: hoveredCounty[0]['sites'],
+          lockdowns: hoveredCounty[0]['lockdowns'],
+          lockdownInfo: hoveredCounty[0]['lockdownInfo'],
+          updated: updatedDate,
+          created: createdDate
+        });
+      });
+
+      map.dragRotate.disable();
+    });
+  }
+
+  public getFipsLockdownData(fipsCode: string) {
+    const lockdownUrl = 'https://nodes.geoservices.tamu.edu/api/covid/lockdowns/active/county/' + fipsCode;
+    const requests = forkJoin([this.http.get(lockdownUrl)]);
+    requests.pipe(take(1)).subscribe(([lockdownData]) => {
+      console.log(lockdownData);
+      return lockdownData;
+    });
+  }
+
+  public getStats(): void {
+    const statsUrl = 'https://nodes.geoservices.tamu.edu/api/covid/counties/stats';
+    const requests = forkJoin([this.http.get(statsUrl)]);
+    requests.pipe(take(1)).subscribe(([recievedData]) => {
+      this.unformattedData = recievedData;
+      this.formatData();
+      this.drawClaimsMap();
+      this.drawLockdownMap();
+    });
+  }
+
+  public formatData(): void {
+    this.statData = new Array();
+    let lockdownRecords = new Array();
+    Object.keys(this.unformattedData).forEach((keyIndex) => {
+      lockdownRecords = this.unformattedData[keyIndex]['lockdownInfo'].filter((item) => {
+        return item !== null;
+      });
+      lockdownRecords = lockdownRecords.map((item) => {
+        const lockdownStat: LockdownRecord = {
+          updated: item.updated,
+          created: item.created,
+          guid: item.guid
+        };
+        return lockdownStat;
+      });
+
+      const statToAdd: StatRecord = {
+        fips: keyIndex,
+        state: keyIndex.substring(0, 2),
+        fipsNum: parseInt(keyIndex, 10),
+        claims: this.unformattedData[keyIndex]['claims'],
+        sites: this.unformattedData[keyIndex]['sites'],
+        lockdowns: this.unformattedData[keyIndex]['lockdowns'],
+        lockdownInfo: lockdownRecords
+      };
+      this.statData.push(statToAdd);
```

Comment:
Set `this.statData` to be this whole operation instead of pushing data to it, using a `map` method instead of a `forEach`.

`forEach` should only need to be used sparingly, when you don't actually want to mutate any data and instead want to call an action against some data. For example, triggering events.

`map` should be used when you have some data and want different data than the input, based on the input.